internal/gatewayapi/route.go

```diff
@@ -1625,17 +1621,25 @@ func getIREndpointsFromEndpointSlice(endpointSlice *discoveryv1.EndpointSlice, p
 	for _, endpoint := range endpointSlice.Endpoints {
 		for _, endpointPort := range endpointSlice.Ports {
 			// Check if the endpoint port matches the service port
-			// and if endpoint is Ready
-			if *endpointPort.Name == portName &&
-				*endpointPort.Protocol == portProtocol &&
-				// Unknown state (nil) should be interpreted as Ready, see https://pkg.go.dev/k8s.io/api/discovery/v1#EndpointConditions
-				(endpoint.Conditions.Ready == nil || *endpoint.Conditions.Ready) {
-				for _, address := range endpoint.Addresses {
-					ep := ir.NewDestEndpoint(
-						address,
-						uint32(*endpointPort.Port))
-					endpoints = append(endpoints, ep)
+			if *endpointPort.Name != portName || *endpointPort.Protocol != portProtocol {
+				continue
+			}
+			conditions := endpoint.Conditions
+			draining := false
+			// Unknown state (nil) should be interpreted as Ready, see https://pkg.go.dev/k8s.io/api/discovery/v1#EndpointConditions
+			switch {
```

Comment:
the `switch` style is reducing the readability of this logic (it may have been needed to pacify the linter  :) ) 
would something like the below snippet be more readable, and correct ?
```
// add ready endpoints
if conditions.Ready == nil || (*endpoint.Conditions.Ready) {
  ep := ...
  eps = append(eps, ep)
}

// Add Draining endpoints - not ready but serving
if (conditions.Ready != nil && !endpoint.Conditions.Ready) && conditions.Serving != nil && endpoint.Conditions.Serving {
  ep := ... with drain
  eps = append(eps, ep)
}

```

Reply:
looks good, you also accounted for an extra case where serving is `true` , but terminating may not be `true` !